File: src/main/kotlin/com/example/acl/domains/users/repositories/UrlAccessRepository.java
package com.example.acl.domains.users.repositories;

import com.example.auth.entities.UrlAccess;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Repository
public interface UrlAccessRepository extends JpaRepository<UrlAccess, Long> {
    @Modifying
    @Query("DELETE FROM UrlAccess u WHERE u.privilege.id=:privilegeId")
    void deleteByPrivilegeId(@Param("privilegeId") Long privilegeId);
}

File: src/main/kotlin/com/example/acl/domains/users/repositories/UrlAccessRepository.kt
package com.example.acl.domains.users.repositories

import org.springframework.data.jpa.repository.JpaRepository
import com.example.auth.entities.UrlAccess
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface UrlAccessRepository : JpaRepository<UrlAccess, Long> {
    @Modifying
    @Query("DELETE FROM UrlAccess u WHERE u.privilege.id=:privilegeId")
    fun deleteByPrivilegeId(@Param("privilegeId") privilegeId: Long)

    @Query("SELECT u FROM UrlAccess u WHERE u.privilege.id=:privilegeId")
    fun findByPrivilege(@Param("privilegeId") privilegeId: Long): List<UrlAccess>
}